src/lib/hdlc/hdlc.hpp

```diff
@@ -33,13 +33,12 @@
 #ifndef HDLC_HPP_
 #define HDLC_HPP_
 
-#include "openthread-core-config.h"
```

Comment:
Can we consider keeping this? (even though I see that we don't use any configs in hdlc implementation would be good for consistency and future use).
We already include the config header in other libs: e.g. `spinel_encoder.hpp` and `ncp_base.hpp`

Thoughts?

Reply:
I didn't see spinel_encoder.hpp includes the core config file. I intended to remove dependency on core config, which seems to be focusing on `src/core` config. Also this ease third party utilization of these libraries.

Reply:
Yes. You are right. I miss-read `"openthread-spinel-config.h"` in `spinel-encoder.hpp`. I forgot you already helped add a spinel specific config. I agree and think not using core config is definitely the right way for all lib/utils modules.